x-pack/plugins/observability_solution/inventory/public/components/grouped_inventory/grouped_entities_grid.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { EuiDataGridSorting } from '@elastic/eui';
+import useEffectOnce from 'react-use/lib/useEffectOnce';
+import { useInventorySearchBarContext } from '../../context/inventory_search_bar_context_provider';
+import { useKibana } from '../../hooks/use_kibana';
+import { EntitiesGrid } from '../entities_grid';
+import {
+  entityPaginationRt,
+  type EntityColumnIds,
+  type EntityType,
+} from '../../../common/entities';
+import { useInventoryAbortableAsync } from '../../hooks/use_inventory_abortable_async';
+import { useInventoryParams } from '../../hooks/use_inventory_params';
+import { useInventoryRouter } from '../../hooks/use_inventory_router';
+import { extractPaginationParameter } from '../../utils/extract_pagination_parameter';
+
+interface Props {
+  field: string;
+}
+
+export function GroupedEntitiesGrid({ field }: Props) {
+  const { query } = useInventoryParams('/');
+  const { sortField, sortDirection, kuery, pagination: paginationQuery } = query;
+  const inventoryRoute = useInventoryRouter();
+  const pagination = extractPaginationParameter(paginationQuery);
+  const pageIndex = pagination?.[field] ?? 0;
+
+  const { refreshSubject$ } = useInventorySearchBarContext();
+  const {
+    services: { inventoryAPIClient },
+  } = useKibana();
+
+  useEffectOnce(() => {
+    const refreshSubscription = refreshSubject$.subscribe(() => refresh());
+
+    return () => refreshSubscription.unsubscribe();
+  });
```

Comment:
Should `useEffectOnce` be called after `useInventoryAbortableAsync`? Asking because `refresh` is initialized by `useInventoryAbortableAsync`